src/utils/helpers/list/buildQueryFilters.ts

```diff
@@ -72,13 +126,20 @@ const inList = (values: string) => ({ inInsensitive: splitCommaList(values) })
 // Use this if the values must conform to an Enum type (e.g. status, outcome)
 const inEnumList = (values: string, enumList: any) => ({
   in: splitCommaList(values)
-    .map((value) => value.toUpperCase())
-    .filter((value) => Object.values(enumList).includes(value)),
+    .map((value) => value.toUpperCase().replace(' ', '_'))
+    .filter((value) => [...Object.values(enumList)].includes(value)),
 })
 
+// Can represent dates as relative numbers (number of days), i.e. -1, +4, 4, -3
+const convertRelativeDates = (dateStrings: string[]) =>
+  dateStrings.map((dateString) => {
+    if (!dateString.trim().match(/^[-+\d]+$/g)) return dateString
+    return datePlusDays(Number(dateString))
+  })
+
 const parseDateString = (dateString: string) => {
```

Comment:
I think the logic for this should be working well - but it's a little difficult to understand what this is doing on the first look. Not sure what this line 83 is doing:
`...getGenericTypes(),`
I'm log it now and see.

But my question here is, how does the Dates logic (after converting to enum like `today`, `yesterday` works? Do we set the URL with these values and then convert into dates to pass to GraphQL?

Reply:
I am not that happy with implementation of genericTypes tbh, I've added some comments to explain it: 27329d470e6e5a7ef217a0f6a942c05f6a9074b8

Also added more descriptive type (so it's more clear what the key and value definitions in `mapQueryToFilterField` are):

0bccde0bb28d19a0775dc8f2a5c730c6a876d246

Reply:
Cooments look great. Thanks